ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -327,6 +327,18 @@ func (r *RayJobReconciler) getSubmitterTemplate(rayJobInstance *rayv1.RayJob, ra
 		r.Log.Info("default submitter template is used")
 	} else {
 		submitterTemplate = *rayJobInstance.Spec.SubmitterPodTemplate.DeepCopy()
+
```

Comment:
1. How about moving the logic to the same location as the `PYTHONUNBUFFERED` environment variable ([code](https://github.com/ray-project/kuberay/blob/33f29f61d6be0809662df2a58bbc605e7256a114/ray-operator/controllers/ray/rayjob_controller.go#L345-L349))?

2. We can always inject the env variable no matter whether users set `SubmitterPodTemplate` by themselves or not.

3. We can only inject `FQ_RAY_IP` into the Ray container, which is always the first container (i.e., `submitterTemplate.Spec.Containers[utils.RayContainerIndex]`) in a Pod in KubeRay. However, we can inject it into other sidecar containers later upon user request.

4. Do we need to check `len(container.Env) == 0`? Can we directly append to the slice instead?

Reply:
Thanks, I updated to always inject the environment variable. I wasn't sure if it made sense to inject the env var in the default submitter template.